frontend/src/Components/HomePage/AddTrip/FlightForm.jsx

```diff
@@ -0,0 +1,258 @@
+import { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import axios from 'axios';
+import './FlightForm.css'
+import FlightResults from './FlightResults';
+import { API_BASE_URL } from '../../../config.js';
+
+
+
+const FlightForm= () => {
+
+    const navigate = useNavigate();
+    const [flightNumber, setFlightNumber] = useState(() => sessionStorage.getItem('flightNumber') || '');
+    const [departureDate, setDepartureDate] = useState(() => sessionStorage.getItem('departureDate') || '');
+    const [carrierCode, setCarrierCode] = useState(() => sessionStorage.getItem('carrierCode') || '');
+    const [flightStatus, setFlightStatus] = useState(() => {
+        const saved = sessionStorage.getItem('flightStatus');
+        return saved ? JSON.parse(saved) : null;
+    });
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [flightAdded, setFlightAdded] = useState(() => sessionStorage.getItem('flightAdded') === 'true');
+    const [isAddingFlight, setIsAddingFlight] = useState(false);
+
+
+
+    useEffect(() => {
+
+        sessionStorage.setItem('flightNumber', flightNumber);
+        sessionStorage.setItem('departureDate', departureDate);
+        sessionStorage.setItem('carrierCode', carrierCode);
+        sessionStorage.setItem('flightStatus', JSON.stringify(flightStatus));
+        sessionStorage.setItem('flightAdded', flightAdded.toString());
```

Comment:
It's nice to see some caching implemented, but I'm just curious about a few things:

1. Why do you want to cache this individual item (new trip just created)? Do you plan to cache all flights for the current user or only the new one? 
2. How do you plan to handle the overall state for all flights? Where do you plan to "store" that information? Like a higher-level component having a list of flights as state and passing the list as props to child components? 

Up-to-you how you want to handle the state management of your app, but I think this approach might need to be updated when you start fetching all trips from the current user.

There are some options to better handle state management like react redux (https://react-redux.js.org/), but for now, I'd suggest you to re-think your state management for current user flights based on what you learnt in codepath.

3. Why did you decide to use session storage instead of local storage?